backend/sirius-components-diagrams-layout/src/test/java/org/eclipse/sirius/components/diagrams/layout/incremental/NodeCreationTests.java

```diff
@@ -0,0 +1,171 @@
+import org.eclipse.sirius.components.diagrams.layout.services.DefaultTestDiagramDescriptionProvider;
+import org.eclipse.sirius.components.diagrams.layout.services.TestDiagramCreationService;
+import org.eclipse.sirius.components.diagrams.layout.services.TestLayoutObjectService;
+import org.eclipse.sirius.components.diagrams.tests.builder.JsonBasedEditingContext;
+import org.eclipse.sirius.components.diagrams.tests.builder.TestLayoutDiagramBuilder;
+import org.eclipse.sirius.components.representations.IRepresentationDescription;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests the creation of a node and the resizing of its parent.
+ *
+ * @author rpage
+ */
+@SuppressWarnings("checkstyle:MultipleStringLiterals")
+public class NodeCreationTests {
+    private TestLayoutObjectService objectService = new TestLayoutObjectService();
+
+    private DefaultTestDiagramDescriptionProvider defaultTestDiagramDescriptionProvider = new DefaultTestDiagramDescriptionProvider(this.objectService);
+
+    @Test
+    public void testNodeCreationNoResizeNeeded() {
+        Position eventCreationPosition = Position.at(10, 10);
+        Diagram diagram = this.createParentNotResizedDiagram();
+        this.createNewNode(diagram, eventCreationPosition);
+    }
+
+    @Test
+    public void testNodeCreationWithResizeNeeded() {
+        Position eventCreationPosition = Position.at(199, 199);
+        Diagram diagram = this.createParentNotResizedDiagram();
+        this.createNewNode(diagram, eventCreationPosition);
+    }
+
+    @Test
+    public void testNodeCreationInResizedParentNoResizeNeeded() {
+        Position eventCreationPosition = Position.at(10, 10);
+        Diagram diagram = this.createParentResizedDiagram();
+        this.createNewNode(diagram, eventCreationPosition);
+    }
+
+    @Test
+    public void testNodeCreationInResizedParentWithResizeNeeded() {
+        Position eventCreationPosition = Position.at(199, 199);
+        Diagram diagram = this.createParentResizedDiagram();
+        this.createNewNode(diagram, eventCreationPosition);
+    }
+
+    private Diagram createParentNotResizedDiagram() {
+        // @formatter:off
+        return TestLayoutDiagramBuilder.diagram("Root") //$NON-NLS-1$
+                .nodes()
+                    .rectangleNode("Parent").at(10, 10).of(200, 200) //$NON-NLS-1$
+                    .and()
+                .and()
+            .build();
+        // @formatter:on
+    }
+
+    private Diagram createParentResizedDiagram() {
+        // @formatter:off
+        return TestLayoutDiagramBuilder.diagram("Root") //$NON-NLS-1$
+                .nodes()
+                    .rectangleNode("Parent").at(10, 10).of(200, 200).customizedProperties(Set.of(CustomizableProperties.Size)) //$NON-NLS-1$
+                    .and()
+                .and()
+            .build();
+        // @formatter:on
+    }
+
+    private void createNewNode(Diagram diagram, Position eventCreationPosition) {
```

Comment:
The createDiagramCreationService and getNode  methods are also used in NodeResizeTest and DiagramLayoutTest. It would be nice to not duplicate this code, but I am not sure where to put it.

Reply:
As discussed directly, this slight copy / paste is not an issue for now